atomic-primops/Data/Atomics.hs

```diff
@@ -349,7 +349,21 @@ casMutVar2 mv tick new = IO$ \st ->
 -- | Memory barrier implemented by the GHC rts (see SMP.h).
 -- writeBarrier :: IO ()
 
-#if !(defined(mingw32_HOST_OS) && __GLASGOW_HASKELL__ < 802)
+#if __GLASGOW_HASKELL__ > 909
+
+foreign import ccall  unsafe "hs_atomics_primops_store_load_barrier" storeLoadBarrier
+  :: IO ()
+
+-- | Memory barrier implemented by the GHC rts (see SMP.h).
```

Comment:
These Haddocks aren't accurate, since this version of the function isn't implemented by the GHC RTS anymore. I'd favor either (1) revising this comment, or (2) just removing it for consistency with `storeLoadBarrier` above, which also doesn't have a comment.